apps/remote_control/lib/lexical/remote_control/code_mod/rename.ex

```diff
@@ -0,0 +1,70 @@
+defmodule Lexical.RemoteControl.CodeMod.Rename do
+  alias Lexical.Ast.Analysis
+  alias Lexical.Document
+  alias Lexical.Document.Position
+  alias Lexical.Document.Range
+  alias Lexical.Protocol.Notifications.DidChange
+  alias Lexical.Protocol.Notifications.DidSave
+  alias Lexical.RemoteControl.Commands
+  alias Lexical.RemoteControl.Progress
+
+  alias __MODULE__
+
+  @spec prepare(Analysis.t(), Position.t()) ::
+          {:ok, {atom(), String.t()}, Range.t()} | {:error, term()}
+  defdelegate prepare(analysis, position), to: Rename.Prepare
+
+  @rename_mappings %{module: Rename.Module}
+
+  @spec rename(Analysis.t(), Position.t(), String.t(), String.t() | nil) ::
+          {:ok, [Document.Changes.t()]} | {:error, term()}
+  def rename(%Analysis{} = analysis, %Position{} = position, new_name, client_name) do
+    with {:ok, {renamable, entity}, range} <- Rename.Prepare.resolve(analysis, position) do
+      rename_module = @rename_mappings[renamable]
```

Comment:
I think it's fine to use `@rename_mappings[renamable]` here, since `Rename.Prepare.resolve` will let us know whether the renamable is supported or not.

Reply:
Got it! In that case, can we use `Map.fetch!` to be explicit that we expect a known key and so any error occurs at that point, instead of below where we try to call a function from the module?

Reply:
Ok